cocos/ui/UIEditBox/UIEditBoxImpl-win32.cpp

```diff
@@ -203,119 +170,146 @@ namespace ui {
     {
         if (_editBoxInputMode == cocos2d::ui::EditBox::InputMode::ANY)
         {
-            this->createMultilineEditCtrl();
+            this->createEditCtrl(false);
         }
         else if (_editBoxInputMode == cocos2d::ui::EditBox::InputMode::NUMERIC
             || _editBoxInputMode == cocos2d::ui::EditBox::InputMode::DECIMAL
             || _editBoxInputMode == cocos2d::ui::EditBox::InputMode::PHONE_NUMBER)
         {
-            this->createSingleLineEditCtrl();
-            ::SetWindowLongW(hwndEdit, GWL_STYLE, ::GetWindowLongW(hwndEdit, GWL_STYLE) | ES_NUMBER);
+            this->createEditCtrl(true);
+            ::SetWindowLongW(_hwndEdit, GWL_STYLE, ::GetWindowLongW(_hwndEdit, GWL_STYLE) | ES_NUMBER);
         }
         else
         {
-            this->createSingleLineEditCtrl();
+            this->createEditCtrl(true);
         }
 
         if (this->_editBoxInputFlag != cocos2d::ui::EditBox::InputFlag::PASSWORD)
         {
-            PostMessage(hwndEdit, EM_SETPASSWORDCHAR, (WPARAM)0, (LPARAM)0);
+            ::PostMessageW(_hwndEdit, EM_SETPASSWORDCHAR, (WPARAM)0, (LPARAM)0);
         }
     }
     void EditBoxImplWin::setNativeInputFlag(EditBox::InputFlag inputFlag)
     {
         if (inputFlag == EditBox::InputFlag::PASSWORD)
         {
-            this->createSingleLineEditCtrl();
+            this->createEditCtrl(true);
         }
-       
         else
         {
             if (_editBoxInputMode != cocos2d::ui::EditBox::InputMode::ANY)
             {
-                this->createSingleLineEditCtrl();
+                this->createEditCtrl(true);
 
                 if (inputFlag == EditBox::InputFlag::INITIAL_CAPS_ALL_CHARACTERS)
                 {
-                    ::SetWindowLongW(hwndEdit, GWL_STYLE, ::GetWindowLongW(hwndEdit, GWL_STYLE) | ES_UPPERCASE);
+                    ::SetWindowLongW(_hwndEdit, GWL_STYLE, ::GetWindowLongW(_hwndEdit, GWL_STYLE) | ES_UPPERCASE);
                 }
                 // Clear the password style
-                PostMessage(hwndEdit, EM_SETPASSWORDCHAR, (WPARAM)0, (LPARAM)0);
+                ::PostMessageW(_hwndEdit, EM_SETPASSWORDCHAR, (WPARAM)0, (LPARAM)0);
             }
         }
     }
+
     void EditBoxImplWin::setNativeReturnType(EditBox::KeyboardReturnType returnType)
     {
         //not implemented yet
     }
+
+    void EditBoxImplWin::setNativeTextHorizontalAlignment(TextHAlignment alignment)
+    {
+        LONG style = ::GetWindowLongW(_hwndEdit, GWL_STYLE);
+        switch (alignment)
```

Comment:
how about this code style, it's more simple and more easy to understand:
```C++
        switch (alignment)
        {
        case TextHAlignment::LEFT:
            style = (style & ~ES_CENTER & ~ES_RIGHT) | ES_LEFT;
            break;
        case TextHAlignment::CENTER:
            style = (style & ~ES_LEFT & ~ES_RIGHT) | ES_CENTER;
            break;
        case TextHAlignment::RIGHT:
            style = (style & ~ES_LEFT & ~ES_CENTER) | ES_RIGHT;
            break;
        }
```

Reply:
That is a good idea